crates/viewer/re_viewer/src/web.rs

```diff
@@ -197,20 +198,23 @@ impl WebHandle {
     /// It is an error to open a channel twice with the same id.
     #[wasm_bindgen]
     pub fn add_receiver(&self, url: &str, follow_if_http: Option<bool>) {
-        let Some(mut app) = self.runner.app_mut::<crate::App>() else {
+        let Some(app) = self.runner.app_mut::<crate::App>() else {
             return;
         };
+
+        // TODO(andreas): should follow_if_http be part of the fragments?
         let follow_if_http = follow_if_http.unwrap_or(false);
-        if let Some(rx) = url_to_receiver(
-            &self.connection_registry,
-            app.egui_ctx.clone(),
+        let select_redap_source_when_loaded = true;
+        if try_open_url_in_viewer(
```

Comment:
This used to repaint, but it no longer does. Should it?

Reply:
I think it should just to be sure, good catch. I used to have repaint direclty inside of `try_open_url_in_viewer`, but there's a case where this causes a deadlock since we're already inside a `egui_ctx` capture and this was kinda hard to catch for me